### PR TITLE
WIP: v4 List Groups

### PR DESCRIPTION
**Pull request by kendaganio**
Issue #3027

TODO
- [x] fix docs
- [x] createBootstrapComponent
- [ ] fade (?)
- [x] variants
- [x] context classes 
- [x] clickable ListGroupItems

### CODE REVIEW

**Comment by jquense** [comment]
hey @kendaganio sorry for the delay, do you need input here? No rush just want to make sure you aren't waiting on us

**Comment by kendaganio** [comment]
Hello @jquense, I haven't completed the entire thing yet. Have been a bit busy at work lately. Will get back to it next week!

**Comment by jquense** [comment]
@kendaganio I've rebased and taken a bit of a shot at finishing this off. Thanks a ton for your work so far!

**Comment by kendaganio** [comment]
terribly sorry @jquense for leaving this hanging for quite some time, was a bit busy at work but will try my best to continue helping out whenever i'm free!

**Comment by jquense** [comment]
no worries! we all get busy, i've also been super short on time

File: src/AbstractNav.js [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+import PropTypes from 'prop-types';
+import elementType from 'prop-types-extra/lib/elementType';
+import mapContextToProps from 'react-context-toolbox/lib/mapContextToProps';
+import SelectableContext, { makeEventKey } from './SelectableContext';
+import NavContext from './NavContext';
+import TabContext from './TabContext';
+
+const noop = () => {};
+
+class AbstractNav extends React.Component {
```
Comment (jquense) [inline src/AbstractNav.js]: cheeeck this out @taion i quite like the sort of composability of the new contexts

**Comment by jquense** [comment]
The ul/li mismatch is even more annoying now that bootstrap upstream doesn't care about what element you use (nave as well). I think that for simple cases a ul/li syrucutre is semantically and a11y the most correct but we cant enforce they are paired together so I think having completely incorrect markup easily is a worse outcome. I think moving away from ul/li as default elements will probably cause less bugs than they create

**Comment by taion** [comment]
i'm totally cool with that – seems fine to ask users to use the correct `as`

**Pull request merged**